python/cudf/cudf/core/column/string.py

```diff
@@ -4506,6 +4506,52 @@ def __init__(
             children=children,
         )
 
+        self._start_offset = None
+        self._end_offset = None
+
+    @property
+    def start_offset(self):
+        if self._start_offset is None:
+            if len(self.base_children) == 2:
+                self._start_offset = int(self.base_children[0][self.offset])
+            else:
+                self._start_offset = 0
+
+        return self._start_offset
+
+    @property
+    def end_offset(self):
+        if self._end_offset is None:
+            if len(self.base_children) == 2:
+                self._end_offset = int(
+                    self.base_children[0][self.offset + self.size]
+                )
```

Comment:
What if the child is 0 sized but there is a column?

Reply:
for example like this?

```python
>>> s = cudf.Series(['a'], dtype='str')
>>> s[1:]
Series([], dtype: object)
>>> s._column.base_children
(<cudf.core.column.numerical.NumericalColumn object at 0x7f23cb301320>, <cudf.core.column.numerical.NumericalColumn object at 0x7f24bedc4f80>)
>>> s[1:]._column.base_children
(<cudf.core.column.numerical.NumericalColumn object at 0x7f24bedf1b90>, <cudf.core.column.numerical.NumericalColumn object at 0x7f24bdac4680>)
```

Reply:
Yup exactly

Reply:
In that case we return the size of `child[0]`. Since both offset and size are 0, the net size of child[1] will become 0:

```python
>>> import cudf
>>> s = cudf.Series(['abc'])
>>> s.memory_usage()
11
>>> s
0    abc
dtype: object
>>> s[1:]
Series([], dtype: object)
>>> s[1:].memory_usage()
4
>>> s[1:].memory_usage()
> /nvme/0/pgali/envs/cudfdev/lib/python3.7/site-packages/cudf/core/column/string.py(4544)__sizeof__()
-> self.end_offset - self.start_offset
(Pdb) child0_size
4
(Pdb) self.end_offset - self.start_offset
0
(Pdb) self.end_offset
0
(Pdb) self.start_offset
0
```

Reply:
Is there ever a situation where we have an empty column for the offsets column?

Reply:
Doesn't look like a possibility seeing the existing code usages:

https://github.com/rapidsai/cudf/blob/84a7c6ac7d5a0261f53312706841fed7f59837d9/python/cudf/cudf/core/column/string.py#L4492-L4497

https://github.com/rapidsai/cudf/blob/84a7c6ac7d5a0261f53312706841fed7f59837d9/python/cudf/cudf/core/column/string.py#L4509-L4517


But I have added a bounds check for both just in case to not break with an index error if we choose to deviate from the way we store offsets in `base_children` currently.